### PR TITLE
Adds filtering to Archives, Algorithms and Reader Studies

### LINKED ISSUE #1649: Add Modality and Structure labels and filters to Archive, Reader Studies and Algorithms
Now that `structures` and `modalities` are in the correct apps we can add m2m relationships to the from Archives, Reader Studies and Algorithms. Add this, and the filters along with filtering on title and description to the list pages. Required for #1191.

`task_type` and `challenge_series` I don't think are required here so they should not be added.

### PR DESCRIPTION
**Pull request by jmsmkn**
This PR adds links from Archives, Algorithms and Reader Studies to Publications, Structures and Modalities, and adds common filter methods for these models and Challenges.

Closes #1649

### CODE REVIEW

File: app/grandchallenge/reader_studies/forms.py [hunk 1]
```diff
@@ -96,6 +99,8 @@ class Meta:
         help_texts = READER_STUDY_HELP_TEXTS
         widgets = {
             "description": TextInput,
+            "modalities": Select2MultipleWidget,
+            "structures": Select2MultipleWidget,
         }
```
Comment (HarmvZ) [inline app/grandchallenge/reader_studies/forms.py:104]: Did you leave out the `publications` field here on purpose?

**Review by HarmvZ — APPROVED** [review]
LGTM. One question

File: app/grandchallenge/reader_studies/forms.py (hunk 1 above)
Comment (jmsmkn) [reply app/grandchallenge/reader_studies/forms.py:104]: No, nice find!

Comment (jmsmkn) [reply app/grandchallenge/reader_studies/forms.py:106]: We really should use the same form for create and update...

**Pull request merged**